extensions/shared/external-media/media-browser/index.js

```diff
@@ -76,10 +82,95 @@ function MediaBrowser( props ) {
 	const prevMediaCount = useRef( 0 );
 
 	const onLoadMoreClick = () => {
-		prevMediaCount.current = media.length;
+		if ( media.length ) {
+			setFocused( media.length );
+		}
 		nextPage();
 	};
 
+	const navigate = ( keyCode, index ) => {
+		switch ( keyCode ) {
+			case LEFT:
+				if ( index >= 1 ) {
+					setFocused( index - 1 );
+				}
+				break;
+			case RIGHT:
+				if ( index < media.length ) {
+					setFocused( index + 1 );
+				}
+				break;
+			case UP:
+				if ( index >= columns.current ) {
+					setFocused( index - columns.current );
+				}
+				break;
+			case DOWN:
+				if ( index < media.length - columns.current ) {
+					setFocused( index + columns.current );
+				}
+				break;
+		}
+	};
+
+	/**
+	 * Counts how many items are in a row by checking how many of the grid's child
+	 * items have a matching offsetTop.
+	 */
+	const checkColumns = () => {
+		let perRow = 1;
+
+		const items = gridEl.current.children;
+
+		if ( items.length > 0 ) {
+			const firstOffset = items[ 0 ].offsetTop;
+
+			/**
+			 * Check how many items have a matching offsetTop. This will give us the
+			 * total number of items in a row.
+			 */
+			while ( perRow < items.length && items[ perRow ].offsetTop === firstOffset ) {
+				++perRow;
+			}
+		}
+
+		columns.current = perRow;
+	};
+
+	const checkColumnsDebounced = debounce( checkColumns, 400 );
+
+	useEffect( () => {
+		// Re-set columns on window resize:
+		window.addEventListener( 'resize', checkColumnsDebounced );
+		return () => {
+			window.removeEventListener( 'resize', checkColumnsDebounced );
+		};
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	useEffect( () => {
+		// Set columns value once when media are loaded.
+		if ( media.length && columns.current === -1 ) {
+			checkColumns();
+		}
+	}, [ media ] );
+
+	const handleMediaItemClick = ( event, { item, index } ) => {
```

Comment:
`event` and `index` aren't used, which trigger an eslint warning.  `index` can be removed easily enough, since it's being destuctured and isn't required, but `event` is trickier since it's a required positional argument.  In my testing, renaming it to `_event` made the warning go away.